File: 1711-find-valid-matrix-given-row-and-column-sums/1711-find-valid-matrix-given-row-and-column-sums.java
class Solution {
    public int[][] restoreMatrix(int[] row, int[] col) {
        int n = row.length, m = col.length;
        int[][] ans = new int[n][m];

        for(int i = 0; i<n; i++) {
            for(int j = 0; j<m; j++) {
                ans[i][j] = Math.min(row[i], col[j]);
                row[i] -= ans[i][j];
                col[j] -= ans[i][j];
            }
        }

        return ans;
    }
}

File: 1711-find-valid-matrix-given-row-and-column-sums/1711-find-valid-matrix-given-row-and-column-sums.py
class Solution(object):
    def restoreMatrix(self, row, col):
        n, m = len(row), len(col)
        ans = [[0] * m for _ in range(n)]

        for i in range(n):
            for j in range(m):
                ans[i][j] = min(col[j], row[i])
                row[i] -= ans[i][j]
                col[j] -= ans[i][j]
            
        return ans